components/webui/imports/ui/SearchView/SearchView.jsx

```diff
@@ -98,13 +107,19 @@ const SearchView = () => {
             ],
         };
 
-        // NOTE: Although we publish and subscribe using the name
-        // `Meteor.settings.public.SearchResultsCollectionName`, the rows are still returned in the
-        // job-specific collection (e.g., "1"); this is because on the server, we're returning a
-        // cursor from the job-specific collection and Meteor creates a collection with the same
-        // name on the client rather than returning the rows in a collection with the published
-        // name.
-        return dbRef.current.getOrCreateCollection(jobId).find({}, findOptions).fetch();
+        if (SEARCH_SIGNAL.RESP_DONE !== resultsMetadata.lastSignal) {
+            // avoid refreshing estimatedNumResults after job is DONE
+            // since the count would have been already available in
+            // `resultsMetadata.numTotalResults`
+            resultsCollection.estimatedDocumentCount()
```

Comment:
Is the promise guaranteed not to `reject`?

Reply:
I can't find any docs from [Meteor](https://docs.meteor.com/api/collections) or the [official MongoDB API Docs](https://www.mongodb.com/docs/manual/reference/method/db.collection.estimatedDocumentCount/) about any Exception / Error that could be raised. 

Still it's a good practice to add `.catch` when the behaviour is unknown. I can imagine when the db connection is down the Promise would reject somewhere.